backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/AggregationEngine.java

```diff
@@ -793,27 +797,45 @@ private void setDocType(@NonNull final InvoiceHeaderImpl invoiceHeader)
 
 		final I_C_DocType docTypeToBeUsed;
 
-		// Case: Multiple DocTypes are aggregated
-		if (invoiceHeader.getAggregatedDocTypeInvoicingPoolId().isPresent())
+		if (invoiceHeader.getDocTypeInvoicingPoolId().isPresent())
 		{
-			final DocTypeInvoicingPool docTypeInvoicingPool = docTypeInvoicingPoolService.getById(invoiceHeader.getAggregatedDocTypeInvoicingPoolId().get());
+			final DocTypeInvoicingPool docTypeInvoicingPool = docTypeInvoicingPoolService.getById(invoiceHeader.getDocTypeInvoicingPoolId().get());
```

Comment:
i guess u are really creating the DocTypeInvoicePool and persisting it to database because you will need somewhere the DocTypeInvoicePoolId so save, right?

Reply:
right, I'm doing it in `de.metas.invoicecandidate.api.impl.AggregationEngine#getOrCreateDocTypeInvoicingPool`, which is called from `de.metas.invoicecandidate.api.impl.AggregationEngine#addToInvoiceHeader` when `icRecord.getC_DocTypeInvoice_ID() > 0`